skxray/speckle_analysis.py

```diff
@@ -0,0 +1,317 @@
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+    This module will provide statistical analysis for the speckle patterns
+"""
+
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+from six.moves import zip
+from six import string_types
+
+import skxray.correlation as corr
+import skxray.roi as roi
+import skxray.core as core
+from scipy.ndimage.measurements import maximum, mean
+
+try:
+    iteritems = dict.iteritems
+except AttributeError:
+    iteritems = dict.items  # python 3
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def roi_max_counts(images_sets, label_array):
+    """
+    This will determine the highest speckle counts occurred in the required
+    ROI's in required images.
+
+    Parameters
+    ----------
+    images_sets : array
+        sets of images as an array
+
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    max_counts : int
+        maximum speckle counts
+    """
+    max_cts = 0
+    for img_set in images_sets:
+        for img in img_set:
+            max_cts = max(max_cts, maximum(img, label_array))
+    return max_cts
+
+
+def roi_pixel_values(image, labels):
+    """
+    This will provide intensities of the ROI's of the labeled array
+    according to the pixel list
+    eg: intensities of the rings of the labeled array
+
+    Parameters
+    ----------
+    image : array
+        image data dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    roi_pix : dict
+        intensities of the ROI's of the labeled array according
+        to the pixel list
+        {ROI 1 : intensities of the pixels of ROI 1, ROI 2 : intensities of
+         the pixels of ROI 2}
+    """
+
+    if labels.shape != image.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+
+    return {n: image[labels == n] for n in range(1, np.max(labels)+1)}
+
+
+def time_series(number=2, number_of_images=50):
+    """
+    This will provide the geometric series for the integration.
+    Last values of the series has to be less than or equal to number
+    of images
+    ex: number_of_images = 100
+    number = 2, time_series =  1, 2, 4, 8, 16, 32, 64
```

Comment:
These are integration times. e.g.: they take images in every 1ms for 100 images, 
In XSVS, they want to see the scattering patterns with longer integration times like 1ms,2ms, 4ms,8ms, 16ms, 32ms, 64ms (Because they have only 100 images they can't go 128ms) or (1ms, 10ms, 100ms like that), 
This function has to create that time series